### PR TITLE
fix(sts): Use the right region to validate credentials

### LINKED ISSUE #2347: [Bug]: Unable to scan AWS China region account
### Steps to Reproduce

1. Running the following command: 
`prowler aws -o /home/prowler-output -p test-china-profile -f cn-north-1 cn-northwest-1 --log-level ERROR --verbose`
2. Running against cloud provider - `AWS`
3. Single account environment
4. Error I am experiencing: 
`2023-05-11 20:15:26,438 [File: audit_info.py:42] 	[Module: audit_info]	 CRITICAL: ClientError -- An error occurred (InvalidClientTokenId) when calling the GetCallerIdentity operation: The security token included in the request is invalid`

### Expected behavior

I am able to successfully configure the AWS credential, however I experience an error when trying to use it with Prowler. Any ideas what I am doing wrong? I noticed in the log file it is using the global sts endpoint but specifying the us-east-1 region. Shouldn't you be able to specify the region sts uses with whatever is passed to the -f flag? 

### Actual Result with Screenshots or Logs

[2023-05-11_debug.log](https://github.com/prowler-cloud/prowler/files/11456205/2023-05-11_debug.log)

### How did you install Prowler?

Cloning the repository from github.com (git clone)

### Environment Resource

Local workstation

### OS used

MacOS and Linux

### Prowler version

3.4.0

### Pip version

22.0.4

### Context

_No response_

### PR DESCRIPTION
**Pull request by jfagoagas**
### Context

Fixes #2347

### Description

The following fixes has been introduce:
- Set an AWS region when calling `sts.get_caller_identity()`
- Handle some missing resources for the ARN parsing library
- Include more test cases
- Refactor:
  - Move `validate_aws_credentials` to the AWS Provider folder.
  - Move `print_aws_credentials` to the AWS Provider folder.
- Improve logging and error handling.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/tutorials/aws/regions-and-partitions.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+aws_access_key_id = XXXXXXXXXXXXXXXXXXX
+aws_secret_access_key = XXXXXXXXXXXXXXXXXXX
+region = us-gov-east-1
+```
+>> With this option all the partition regions will be scanned without the need of use the `-f/--region` flag
+
+
+## AWS ISO (US & Europe)
+
+For the AWS ISO partitions, which are known as "secret partitions" and are air-gapped from the internet there is no builtin way to scanned it. In this scenario if you want to audit an AWS Account in one of the AWS ISO partitions you should manually update the [aws_regions_by_service.json](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/aws/aws_regions_by_service.json) and include the partition, region and services, e.g.:
```
Comment (jfagoagas) [inline docs/tutorials/aws/regions-and-partitions.md:51]:
```suggestion
For the AWS ISO partitions, which are known as "secret partitions" and are air-gapped from the internet there is no builtin way to scan it. In this scenario if you want to audit an AWS Account in one of the AWS ISO partitions you should manually update the [aws_regions_by_service.json](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/aws/aws_regions_by_service.json) and include the partition, region and services, e.g.:
```

File: docs/tutorials/aws/regions-and-partitions.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+prowler aws --region us-gov-east-1 us-gov-west-1
+```
+- Using the region configured in your AWS profile at `~/.aws/credentials` or `~/.aws/config`:
+```
+[default]
+aws_access_key_id = XXXXXXXXXXXXXXXXXXX
+aws_secret_access_key = XXXXXXXXXXXXXXXXXXX
+region = us-gov-east-1
+```
+>> With this option all the partition regions will be scanned without the need of use the `-f/--region` flag
```
Comment (jfagoagas) [inline docs/tutorials/aws/regions-and-partitions.md]:
```suggestion
> With this option all the partition regions will be scanned without the need of use the `-f/--region` flag
```

File: mkdocs.yml [hunk 3]
```diff
@@ -43,6 +43,7 @@ nav:
           - Assume Role: tutorials/aws/role-assumption.md
           - AWS Security Hub: tutorials/aws/securityhub.md
           - AWS Organizations: tutorials/aws/organizations.md
+          - Regions and Partition: tutorials/aws/regions-and-partitions.md
```
Comment (MrCloudSec) [inline mkdocs.yml]:
```suggestion
          - AWS Regions and Partitions: tutorials/aws/regions-and-partitions.md
```

File: prowler/providers/aws/lib/arn/arn.py [hunk 4]
```diff
@@ -9,17 +9,17 @@
...
 )
 
 
-def arn_parsing(arn):
+def parse_iam_credentials_arn(arn):
     # check for number of fields, must be six
     if len(arn.split(":")) != 6:
         raise RoleArnParsingFailedMissingFields
     else:
         arn_parsed = arnparse(arn)
```
Comment (MrCloudSec) [inline prowler/providers/aws/lib/arn/arn.py]: can you remove `arnparse` library?

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, review my comments.

File: prowler/providers/aws/lib/arn/arn.py (hunk 4 above)
Comment (MrCloudSec) [reply prowler/providers/aws/lib/arn/arn.py]: Done

**Review by MrCloudSec — DISMISSED** [review]

**Review by MrCloudSec — APPROVED** [review]
amazing 🤯

**Pull request merged**